src/Microsoft.ML.Data/DataLoadSave/Binary/Codecs.cs

```diff
@@ -1276,7 +1272,112 @@ private bool GetKeyCodec(Stream definitionStream, out IValueCodec codec)
                 Contracts.CheckDecode((ulong)count <= itemType.RawKind.ToMaxInt());
                 Contracts.CheckDecode(contiguous || count == 0);
 
-                type = new KeyType(itemType.RawKind, min, count, contiguous);
+                type = new KeyType(itemType.RawType, count);
```

Comment:
>count [](start = 53, length = 5)

Let's imagine, hypothetically, that the count were 0, that is, it is of unknown cardinality. In such an instance as that, I would not like this to fail. It seems like the only reasonable course would be that we map it to the greatest possible value... so if U1, we put in `byte.MaxValue`, if U2, then `ushort.MaxValue`, etc.? This would be a suitable enough mapping. If we get rid of the concept of "unknown count," then I'd rather not fail, I'd rather just say, "you know what? I know I know nothing" and allow it to be everything. :) Which seems less harmful.